sql-cli/src/opensearch_sql_cli/formatter.py

```diff
@@ -55,10 +55,15 @@ def format_output(self, data):
         formatter = TabularOutputFormatter(format_name=self.table_format)
 
         # parse response data
-        datarows = data["datarows"]
-        schema = data["schema"]
-        total_hits = data["total"]
-        cur_size = data["size"]
+        try:
+            datarows = data["datarows"]
+            schema = data["schema"]
+            total_hits = data["total"]
+            cur_size = data["size"]
+        except KeyError as e:
+            # in case some errors in query engine don't throw exception, leading to a KeyError
+            raise Exception(data)
```

Comment:
is there a more specific exception type that can be used? if not this is fine

Reply:
use `OpenSearchException` instead. Also created issue #311 for query engine, I think engine should handle this correctly and throw some meaningful exception